src/main/java/org/broadinstitute/hellbender/utils/SimpleInterval.java

```diff
@@ -229,6 +230,8 @@ public boolean overlaps( final Locatable other ) {
       * @return true if this interval overlaps other, otherwise false
       */
      public boolean overlapsWithMargin(final Locatable other, final int margin) {
+         ParamUtils.isPositiveOrZero(margin,
+                 "given margin is negative: " + margin + "\tfor this: " + toString() + "\tand that: " + other.toString());
```

Comment:
This will end up calling `toString()` on both intervals on every call to this method, even if margin is >= 0. Since this is the kind of method that is called millions of times, this could be a performance issue.

 You should refactor this change so that the `toString() calls only happen if there's an actual error and you are going to throw.